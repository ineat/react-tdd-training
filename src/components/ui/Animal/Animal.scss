.animal-container {
  position: relative;
  cursor: pointer;

  &:first-of-type {
    margin-right: 2rem;
  }

  &:last-of-type {
    margin-left: 2rem;
  }

  .description {
    position: absolute;
    bottom: -20px;
    left: 0;
    z-index: 1;
    display: block;
    width: 350px;
    margin: 0;
    padding: 0;
    color: #444;
    font-size: 25px;
    font-weight: bold;
    text-decoration: none;
    text-align: center;
    -webkit-transition: .3s ease-in-out;
    transition: .3s ease-in-out;
    opacity: 0;
  }

  .animal {
    width: 350px;
    border-radius: 30px;

    -webkit-filter: sepia(100%);
    filter: sepia(100%);
    -webkit-transition: .3s ease-in-out;
    transition: .3s ease-in-out;

    &:hover {
      -webkit-filter: sepia(0);
      filter: sepia(0);
    }
    &:hover+span {
      bottom: 25px;
      opacity: 1;
    }
  }
}