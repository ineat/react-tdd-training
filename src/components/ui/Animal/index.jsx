import * as React from 'react';

import './Animal.scss';

export default ({ image, alt, description }) => (
  <div className="animal-container">
    <img className="animal" src={image} alt={alt}/>
    <span className="description">
      {description}
    </span>
  </div>
);