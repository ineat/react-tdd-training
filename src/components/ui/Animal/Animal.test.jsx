import * as React from 'react';
import { mount, shallow } from 'enzyme';

import Animal from '.';

describe('Animal', () => {
  const onSelectAnimalMock = jest.fn();
  const props = {
    image: 'dat image',
    alt: 'Arg arg!',
    description: 'I love dat',
    onSelectAnimal: () => onSelectAnimalMock('Arg')
  };

  it('Should render.', () => {
    expect(mount(<Animal {...props} />)).toMatchSnapshot();
  });

  describe('Check props', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Animal {...props} />);
    });

    it('Should call "onSelectAnimal" when clicking on it.', () => {
      wrapper.find('.animal').simulate('click');

      expect(onSelectAnimalMock).toHaveBeenCalled();
    });

    it('Should display the description.', () => {
      const text = wrapper.find('.description').text();

      expect(text).toEqual(props.description);
    });

    it('Should display the image by using given image and given alt text.', () => {
      const { src, alt } = wrapper.find('.animal').props();

      expect(src).toEqual(props.image);
      expect(alt).toEqual(props.alt);
    });
  });
});