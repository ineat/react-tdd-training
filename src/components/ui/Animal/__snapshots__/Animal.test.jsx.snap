// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Should render. 1`] = `
<_default
  alt="Arg arg!"
  description="I love dat"
  img="dat image"
  onSelectAnimal={[Function]}
>
  <div
    className="animal-container"
  >
    <img
      alt="Arg arg!"
      className="animal"
      onClick={[Function]}
    />
    <span
      className="description"
    >
      I love dat
    </span>
  </div>
</_default>
`;
